p2p/node.go

```diff
@@ -390,10 +397,94 @@ func (n *Node) Start() error {
 		}
 	}
 
-	// Advertise ourselves for the purposes of peer discovery.
-	discovery.Advertise(n.ctx, n.routingDiscovery, n.config.RendezvousString, discovery.TTL(advertiseTTL))
+	// Immediately attempt to connect to some peers at the rendezvous points.
+	go func() {
+		if err := n.findNewPeers(n.ctx); err != nil {
+			// Note(albrow): we can just log this error. The peer discovery loop
+			// will keep working in the background so we will automatically try
+			// again.
+			log.WithError(err).Error("initial peer discovery failed")
+		}
+	}()
+
+	// Create a child context so that we can preemptively cancel if there is an
+	// error.
+	innerCtx, cancel := context.WithCancel(n.ctx)
+	defer cancel()
+
+	// Below, we will start several independent goroutines. We use separate
+	// channels to communicate errors and a waitgroup to wait for all goroutines
+	// to exit.
+	wg := &sync.WaitGroup{}
+
+	// Start advertising after a delay.
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		select {
+		case <-innerCtx.Done():
+			// If the context was canceled, return immediately. Don't bother
+			// advertising ourselves.
+			return
+		case <-time.After(advertiseDelay):
+			// Otherwise, advertise ourselves on the DHT after a delay.
+			// The delay allows us to prioritize connecting to peers with a matching
+			// rendezvous point in order of preference.
+			//
+			// Note(albrow): Advertise doesn't return an error, so we have no
```

Comment:
Hypothetically, we could wrap the `Advertise` function and actually test to verify that `Advertise` was successful if this in important.

Reply:
How would that work exactly?